app/src/main/java/sopt/org/millie/data/repositoryimpl/BookDetailRepositoryImpl.kt

```diff
@@ -0,0 +1,20 @@
+package sopt.org.millie.data.repositoryimpl
+
+import sopt.org.millie.data.datasource.BookDetailDataSource
+import sopt.org.millie.data.model.BookDetailModel
+import sopt.org.millie.data.model.toModel
+import sopt.org.millie.data.repository.BookDetailRepository
+import javax.inject.Inject
+
+class BookDetailRepositoryImpl
+    @Inject
+    constructor(
+    private val bookDetailDataSource: BookDetailDataSource,
+) : BookDetailRepository {
+    override suspend fun getBookDetailInformation(bookId: Long): Result<BookDetailModel> {
+        return runCatching {
+           val response = bookDetailDataSource.getBookDetailInformation(bookId)
+            response.data?.toModel() ?: throw IllegalArgumentException("response data is null")
+        }
+    }
```

Comment:
헐 언니 나도 이 코드 보고 생각낫는데 나도 이거 수정 안햇다... 이거 suspendRunCatching 확장 함수 있어서 

<img width="569" height="269" alt="image" src="https://github.com/user-attachments/assets/a84bb430-43a6-498c-895b-a929f7edaeeb" />

이거 쓰면 
```suggestion
    override suspend fun getBookDetailInformation(bookId: Long): Result<BookDetailModel> = suspendRunCatching {
        bookDetailDataSource.getBookDetailInformation(bookId).data!!.toModel()
    }
```
이렇게 더 깔끔띠하게 수정 가능해욧 !!!! suspendRunCatching을 사용하는 이유는 코루틴 취소를 안전하게 처리하기 위해서인데요 !!! 일반 runCatching은 모든 예외를 잡아버리는데 여기에는 저희 저번 세미나에서 들었던 코루틴 취소 신호인 CancellationException도 포함돼용 이 취소 신호를 잡아버리면 사용자가 화면을 나갔는데도 네트워크 요청이 계속 실행되고 이미 사라진 화면의 상태를 업데이트하려고 시도하면서 메모리 누수나 크래시가 발생할 수 있어요...!!! suspendRunCatching은CancellationException을 다시 throw해서 코루틴 취소가 정상적으로 작동하도록 해요이 !! 네트워크 에러 같은 진짜 예외만 Result.failure로 처리하고 취소 신호는 위로 전파시켜서 코루틴이 제대로 종료되게 만드는 거죠 !!! 요거 사용해보는 건 어떨까요 ?? 나도 지금 이슈 파서 수정해야겟다 omg

Reply:
오오 좋아요!!
suspendRunCatching에 대해 좀 더 공부해봐야겠네요~!!

Reply:
@nhyeonii 아 근데 추가로 요기 부분 !! 처리 보다는 ?:는 유지하는 편이 더 나을 것 같은데 어떻게 생각하세요??

Reply:
옹 그 방법도 괜찮을 것 같아요 ㅎㅎㅎㅎ
data가 null이면 NullPointerException이 나긴 하는데 이게 suspendRunCatching 블록 안에서 발생하니까 자동으로 Result.failure로 처리돼서 크래시는 안 나겟지만 !!!!!!!!!!
그래도 의도를 명확하게 드러내는 게 더 좋다는 생각은 저도 동의합니닿ㅎㅎㅎㅎ 언니가 말해준 방식대로 적용해도 넘 좋을 것 같아용 !!

Reply:
넵! 의견 너무 감사합니다!